client/my-sites/patterns/controller.tsx

```diff
@@ -1,16 +1,39 @@
+import wpcom from 'calypso/lib/wp';
 import Patterns from 'calypso/my-sites/patterns/patterns';
 import { getCurrentUserLocale } from 'calypso/state/current-user/selectors';
 import type { Context as PageJSContext } from '@automattic/calypso-router';
 
 type Next = ( error?: Error ) => void;
 
 export function fetchPatterns( context: PageJSContext, next: Next ) {
-	const locale = getCurrentUserLocale( context.store.getState() ) || context.lang || 'en';
+	const { queryClient, lang, store, cachedMarkup } = context;
 
-	// eslint-disable-next-line no-console
-	console.log( '############', locale );
+	if ( cachedMarkup ) {
+		next();
 
-	next();
+		return;
+	}
+
+	const locale = getCurrentUserLocale( store.getState() ) || lang || 'en';
+
+	queryClient
+		.fetchQuery( {
+			queryKey: [ 'patterns', 'library', locale ],
+			queryFn: () => {
+				return wpcom.req.get( `/ptk/patterns/${ locale }`, {
+					categories: 'intro',
+					per_page: '4',
+					post_type: 'wp_block',
+				} );
+			},
+			staleTime: Infinity,
```

Comment:
We should consider `staleTime` closely. If we want to use `Infinity`, then we should probably include all request params in the query key to make sure that any changes to the request URL actually triggers a new data fetch (I ran into trouble with this during testing)

Reply:
Good point, I'll add the category to the query key 👍.